src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,856 @@
+		{
+			DependencyProperty property = args.Property;
+			if (property == TreeViewNode.IsExpandedProperty)
+			{
+				TreeViewNodeIsExpandedPropertyChanged(sender, args);
+			}
+			else if (property == TreeViewNode.HasChildrenProperty)
+			{
+				TreeViewNodeHasChildrenPropertyChanged(sender, args);
+			}
+		}
+
+		private void TreeViewNodeIsExpandedPropertyChanged(TreeViewNode sender, DependencyPropertyChangedEventArgs args)
+		{
+			var targetNode = (TreeViewNode)sender;
+			if (targetNode.IsExpanded)
+			{
+				if (targetNode.Children.Count != 0)
+				{
+					int openedDescendantOffset = 0;
+					IndexOfNode(targetNode, out var index);
+					index = index + 1;
+					for (var i = 0; i < targetNode.Children.Count; i++)
+					{
+						TreeViewNode childNode = null;
+						childNode = (TreeViewNode)targetNode.Children[i];
+						AddNodeToView(childNode, index + i + openedDescendantOffset);
+						openedDescendantOffset = AddNodeDescendantsToView(childNode, index + i, openedDescendantOffset);
+					}
+				}
+
+				//Notify TreeView that a node is being expanded.
+				NodeExpanding?.Invoke(targetNode, null);
+			}
+			else
+			{
+				for (var i = 0; i < targetNode.Children.Count; i++)
+				{
+					TreeViewNode childNode = null;
+					childNode = (TreeViewNode)targetNode.Children[i];
+					RemoveNodeAndDescendantsFromView(childNode);
+				}
+
+				//Notife TreeView that a node is being collapsed
+				NodeCollapsed?.Invoke(targetNode, null);
+			}
+		}
+
+		private void TreeViewNodeHasChildrenPropertyChanged(TreeViewNode sender, DependencyPropertyChangedEventArgs args)
+		{
+			if (m_TreeViewList != null)
+			{
+				var targetNode = sender;
+				var container = m_TreeViewList.ContainerFromNode(targetNode);
+				if (container != null)
+				{
+					TreeViewItem targetItem = (TreeViewItem)container;
+					targetItem.GlyphOpacity = targetNode.HasChildren ? 1.0 : 0.0;
+				}
+			}
+		}
+
+		public bool IsContentMode { get => m_isContentMode; set => m_isContentMode = value; }
+
+		private void ClearEventTokenVectors()
+		{
+			// Remove ChildrenChanged and ExpandedChanged events
+			var inner = GetVectorInnerImpl();
+			for (var i = 0; i < Size; i++)
+			{
+				if (i < inner.Count)
+				{
+					var current = inner[i];
+					var tvnCurrent = (TreeViewNode)current;
+					//TODO
+					//tvnCurrent.ChildrenChanged(m_collectionChangedEventTokenVector.at(i));
+					//tvnCurrent.RemoveExpandedChanged(m_IsExpandedChangedEventTokenVector.at(i));
+				}
+			}
+
+			// Remove SelectedNodeChildrenChangtedEvent
+			var selectedNodes = m_selectedNodes;
+			if (selectedNodes != null)
+			{
+				for (var i = 0; i < selectedNodes.Count; i++)
+				{
+					var current = selectedNodes[i];
+					if (current != null)
+					{
+						var node = (TreeViewNode)current;
+						//TODO
+						//node.ChildrenChanged(m_selectedNodeChildrenChangedEventTokenVector[i]);
+					}
+				}
+			}
+
+			// Clear token vectors
+			//TODO
+			//m_collectionChangedEventTokenVector.clear();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)